kernel/packages/shared/loading/ReportFatalError.ts

```diff
@@ -109,27 +109,34 @@ export async function ReportFatalErrorWithUnityPayloadAsync(error: Error, contex
   }
 }
 
-export function ReportFatalError(error: Error, context: ErrorContextTypes, payload: any = null) {
+export function ReportFatalError(error: Error, context: ErrorContextTypes, payload: Record<string, any> = {}) {
   const finalPayload = GetErrorPayload(context, payload)
-  trackEvent('error_fatal', {
-    context: context,
-    message: error.message,
-    stack: error.stack?.slice(0, 10000)
-  })
+  const eventData = {
+    // attach every field from the payload to the event
+    ...(finalPayload || {}),
```

Comment:
`finalPayload` may contain the Unity payload that's too big for sending it to segment. We should parametrize this method to send only the payloads on the saga stack case.

Reply:
What does this finalpayload thing mean? What is a typical payload? Segment will create a column in the table for each property sent, so i'd like to know what we're going to send here.

Reply:
Addressed in last commit. please check, it is all the `eventData` keys plus all the `payload` keys